src/libc/src/memcmp.c

```diff
@@ -0,0 +1,21 @@
+#include <string.h>
+#include <sys/types.h>
+
+/* gcc is broken and has a non-SUSv2 compliant internal prototype.
+ * This causes it to warn about a type mismatch here.  Ignore it. */
+int memcmp(const void *dst, const void *src, size_t count) {
+  register int r;
+  register const unsigned char *d = dst;
+  register const unsigned char *s = src;
+  ++count;
+  while (--count) {
+    // TODO: check this change !
+    r = (*d - *s);
+    if (r) {
+      return r;
+    }
+    ++d;
+    ++s;
+  }
```

Comment:
@filipeom @chambart 

The original code was:

```c
  while (--count) {
    if (r = (*d - *s))
      return r;
    ++d;
    ++s;
  }
```

But it was raising annoying warnings saying we should add parentheses. I guess it was suggesting using `if ( (r = (*d - *s)) )`, I tried to make the code even more clearer. Could one of you confirm it is correct ? It's been a very long time I had to write C. :-)

Reply:
Looks equivalent to me :smiley: